parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java

```diff
@@ -379,4 +379,32 @@ public long hash(float value) {
   public long hash(Binary value) {
     return hashFunction.hashBytes(value.getBytes());
   }
+
+  private boolean checkCompatibility(BlockSplitBloomFilter that) {
+    return this.maximumBytes == that.maximumBytes &&
+      this.hashStrategy == that.hashStrategy &&
+      this.getAlgorithm() == that.getAlgorithm();
+  }
+
+  @Override
+  public BloomFilter union(BloomFilter bloomFilter) {
+    if (bloomFilter instanceof BlockSplitBloomFilter) {
+      BlockSplitBloomFilter that = (BlockSplitBloomFilter) bloomFilter;
+
+      if (!checkCompatibility(that)) {
+        throw new RuntimeException("Bloom filters are not compatible! Please check"
+          + "if they have the same maximum Bytes, hash strategy and algorithm.");
+      }
+
+      for (int i = 0; i < this.bitset.length; i++) {
+        this.bitset[i] |= that.bitset[i];
+      }
+
+      return new BlockSplitBloomFilter(this.bitset);
+    } else {
+      throw new RuntimeException("Bloom filter should be BlockSplitFilter");
```

Comment:
I know currently we only have `BlockSplitBloomFilter` but later on we might have others. Do we want to support different bloom filter implementation in one parquet file? If yes, do we want to support unioning different implementations?
I think, we shall document it properly in the interface and throw the exception accordingly. For example
```java
throw new IllegalArgumentException("Unsupported BloomFilter implementation to union with: " + bloomFilter.getClass());
```
or
```java
throw new UnsupportedOperationException("Unioning BloomFilter with type " + bloomFilter.getClass() + " is not supported yet");
```

Reply:
I agree!